pkg/client/client.go

```diff
@@ -0,0 +1,131 @@
+	baseUrl, err := url.Parse(base)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{BaseUrl: baseUrl}, nil
+}
+
+type Client struct {
+	BaseUrl *url.URL
+}
+
+// Catalog fetches the service catalog
+func (client *Client) Catalog() *BrokerResponse {
+	return client.makeRequest(http.MethodGet, "catalog", nil)
+}
+
+// Provision creates a new service with the given instanceId, of type serviceId,
+// from the plan planId, with additional details provisioningDetails
+func (client *Client) Provision(instanceId, serviceId, planId string, provisioningDetails json.RawMessage) *BrokerResponse {
+	url := fmt.Sprintf("service_instances/%s?accepts_incomplete=true", instanceId)
+
+	return client.makeRequest(http.MethodPut, url, brokerapi.ProvisionDetails{
+		ServiceID:     serviceId,
+		PlanID:        planId,
+		RawParameters: provisioningDetails,
+	})
+}
+
+// Deprovision destroys a service instance of type instanceId
+func (client *Client) Deprovision(instanceId, serviceId, planId string) *BrokerResponse {
+	url := fmt.Sprintf("service_instances/%s?accepts_incomplete=true&service_id=%s&plan_id=%s", instanceId, serviceId, planId)
+
+	return client.makeRequest(http.MethodDelete, url, nil)
+}
+
+// Bind creates an account identified by bindingId and gives it access to instanceId
+func (client *Client) Bind(instanceId, bindingId, serviceId, planId string, parameters json.RawMessage) *BrokerResponse {
+	url := fmt.Sprintf("service_instances/%s/service_bindings/%s", instanceId, bindingId)
+
+	return client.makeRequest(http.MethodPut, url, brokerapi.BindDetails{
+		ServiceID:     serviceId,
+		PlanID:        planId,
+		RawParameters: parameters,
+	})
+}
+
+// Unbind destroys an account identified by bindingId
+func (client *Client) Unbind(instanceId, bindingId, serviceId, planId string) *BrokerResponse {
+	url := fmt.Sprintf("service_instances/%s/service_bindings/%s?service_id=%s&plan_id=%s", instanceId, bindingId, serviceId, planId)
+
+	return client.makeRequest(http.MethodDelete, url, nil)
+}
+
+func (client *Client) makeRequest(method, path string, body interface{}) *BrokerResponse {
+	br := BrokerResponse{}
+
+	req, err := client.newRequest(method, path, body)
+	br.UpdateRequest(req)
+	br.UpdateError(err)
+	if br.InError() {
+		return &br
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+
+	br.UpdateResponse(resp)
+	br.UpdateError(err)
+
+	return &br
+}
+
+func (client *Client) newRequest(method, path string, body interface{}) (*http.Request, error) {
+	url, err := client.BaseUrl.Parse(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var buffer io.ReadWriter
+	if body != nil {
+		buffer = new(bytes.Buffer)
+		enc := json.NewEncoder(buffer)
+		enc.SetEscapeHTML(false)
+		err := enc.Encode(body)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	request, err := http.NewRequest(method, url.String(), buffer)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("X-Broker-Api-Version", "2.13")
```

Comment:
Would it make sense to promote the version somewhere more prominently located? I could see this getting overlooked during a release as is. Also, if not already, consider adding a note to the documentation mentioning this needs to be modified with each new version.